src/main/java/ac/grim/grimac/checks/Check.java

```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
         return isEnabled && !player.disableGrim && !player.noModifyPacketPermission && !exempted;
     }
 
+    public void updateExempted(){
+        if(player.bukkitPlayer == null || checkName == null) return;
+        FoliaScheduler.getEntityScheduler().run(player.bukkitPlayer, GrimAPI.INSTANCE.getPlugin(),
+                t -> exempted = player.bukkitPlayer != null && player.bukkitPlayer.hasPermission("grim.exempt." + checkName.toLowerCase()),
```

Comment:
You already checked if bukkitPlayer was null
```suggestion
                t -> exempted = player.bukkitPlayer.hasPermission("grim.exempt." + checkName.toLowerCase()),
```